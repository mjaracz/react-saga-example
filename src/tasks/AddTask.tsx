import React, {FC} from 'react'
import { Backdrop, Fade, Modal, TextField, IconButton } from '@material-ui/core'
import { AddCircleOutline } from '@material-ui/icons'
import { useAddTask } from './hooks/useAddTask'
import './styles/add-task.css'
import { IAddTaskProps } from './interfaces/add-task.interface'

export const AddTask: FC<IAddTaskProps> = ({ isOpen, handleAddTaskIcon, setIsOpenModal }) => {
  const { handleSetTitleTask, handleSetDescriptionTask, handleAddTask, titleIsRequired } = useAddTask(setIsOpenModal)
  return (
    <Modal
      aria-labelledby="transition-modal-title"
      aria-describedby="transition-modal-description"
      className="tasks__modal"
      open={isOpen}
      onClose={handleAddTaskIcon}
      closeAfterTransition
      BackdropComponent={Backdrop}
      BackdropProps={{
        timeout: 500,
      }}
    >
      <Fade in={isOpen}>
        <div className="modal__content">
          <TextField className="content__textField" label="Task Title" onChange={handleSetTitleTask} error={titleIsRequired} helperText={ titleIsRequired ? 'title of task is required' : null } required />
          <TextField className="content__textField" label="Task Description" onChange={handleSetDescriptionTask} />
          <IconButton onClick={handleAddTask}>
            <AddCircleOutline/>
          </IconButton>
        </div>
      </Fade>
    </Modal>
  )
}
