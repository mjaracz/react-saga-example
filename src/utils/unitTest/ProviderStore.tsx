import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

export const ProviderStore: FC = ({ children }) => {
  return (
    <Provider store={store}>
      { children }
    </Provider>
  )
}
