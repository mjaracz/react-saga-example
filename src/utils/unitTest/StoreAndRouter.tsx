import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../redux/store'

export const StoreAndRouter: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        { children }
      </BrowserRouter>
    </Provider>
  )
}
