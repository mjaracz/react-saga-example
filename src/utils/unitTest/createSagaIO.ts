import { stdChannel } from 'redux-saga'
import { EventEmitter } from 'events'


export const createSagaIO = (emitter: EventEmitter) => {
  const channel = stdChannel()
  emitter.on('action', channel.put)

  return {
    channel,
    dispatch: action => emitter.emit('action', action),
    getState: () => ({})
  }
}
