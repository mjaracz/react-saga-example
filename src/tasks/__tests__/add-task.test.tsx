import React from 'react'
import { render } from '@testing-library/react'
import { AddTask } from '../AddTask'
import { StoreAndRouter } from '../../utils/unitTest/StoreAndRouter'

describe('<AddTask />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <StoreAndRouter>
        <AddTask handleAddTaskIcon={jest.fn()} isOpen={true} setIsOpenModal={jest.fn()}/>
      </StoreAndRouter>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
