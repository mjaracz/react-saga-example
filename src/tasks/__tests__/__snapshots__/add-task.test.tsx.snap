// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTask /> should match the snapshot 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    aria-describedby="transition-modal-description"
    aria-labelledby="transition-modal-title"
    class="tasks__modal"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="modal__content"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiFormControl-root MuiTextField-root content__textField"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Task Title
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
            â€‰
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root content__textField"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Task Description
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
